src/java/com/rapleaf/jack/AbstractDatabaseModel.java

```diff
@@ -608,43 +607,124 @@ public void clearForeignKeyCache() {
   protected abstract void setAttrs(T model, PreparedStatement stmt)
       throws SQLException;
 
-  @Override
-  public boolean save(T model) throws IOException {
+  // Placeholder default ActiveRecord lock field name
+  private String getLockFieldName(T model) {
+    return "lock_version";
+  }
+
+  // ActiveRecord documentation says it supports optimistic locking using integer fields
+  // That corresponds to the Integer and Long fields, but we only support locking on Integer for now
+  private boolean supportsOptimisticLocking(T model) {
+    final String lockFieldName = getLockFieldName(model);
+
+    if (model.hasField(lockFieldName)) {
+      Object field = model.getField(lockFieldName);
+      return (field != null && field.getClass().equals(Integer.class));
+    } else {
+      return false;
+    }
+  }
```

Comment:
Ah, hmm, perhaps it should be a field in the instance. ActiveRecord supports choosing a different name instead of `lock_version` and I wanted to make that easy to support, but I guess we could make that happen when/if that's supported in Jack.